### PR TITLE
[MO] Result edges naming issue

### PR DESCRIPTION
**Pull request by kczugal**
### Details:
 - when processing a model with multiple outputs, their custom names are not transferred properly because each Result edge fw_tensor_debug_info is overridden by the fw_tensor_debug_info of the first found fake output.

### Tickets:
 - 61417

### CODE REVIEW

**Review by jiwaszki — APPROVED** [review]
Please correct my approval if there is a case when it breaks MO.

File: tools/mo/openvino/tools/mo/front/output_cut.py [hunk 1]
```diff
@@ -35,9 +36,15 @@ def find_and_replace_pattern(self, graph: Graph):
                     if fw_info_value:
                         fw_info = fw_info_value
                         break
+
+            for out_idx in in_node.out_nodes():
+                out_node = in_node.out_node(out_idx)
+                out_nodes_ids[out_idx] = out_node.id
```
Comment (jane-intel) [inline tools/mo/openvino/tools/mo/front/output_cut.py:42]: 'NoneType' object does not support item assignment

**Review by jane-intel — CHANGES_REQUESTED** [review]
It doesn't actually work. All the E2Es failed

File: tools/mo/openvino/tools/mo/front/output_cut.py (hunk 1 above)
Comment (kczugal) [reply tools/mo/openvino/tools/mo/front/output_cut.py:42]: Changed to dict.

**Review by jane-intel — APPROVED** [review]
Asked @rkazants to have a look also

File: tools/mo/openvino/tools/mo/front/output_cut.py [hunk 2]
```diff
@@ -35,9 +36,15 @@ def find_and_replace_pattern(self, graph: Graph):
...
+                for out_idx in in_node.out_nodes():
+                    out_node = in_node.out_node(out_idx)
+                    out_nodes_ids[out_idx] = out_node.id
+            
             graph.erase_node(node)
 
             if fw_info is not None and in_node is not None:
                 for out_idx in in_node.out_nodes():
-                    set_edge_attribute_between_nodes(in_node, in_node.out_node(out_idx),
+                    if node.id == out_nodes_ids[out_idx]:
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/output_cut.py:48]: should not we save `node.id` value in `node_id` due to `graph.erase_node(node)` call before?

**Review by rkazants — APPROVED** [review]

File: tools/mo/openvino/tools/mo/front/output_cut.py (hunk 2 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/front/output_cut.py:48]: Ok, thanks!

**Pull request merged**